packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -192,6 +196,30 @@ export class WorkspaceCommandContribution implements CommandContribution {
                 }
             })
         }));
+        registry.registerCommand(WorkspaceCommands.FILE_DUPLICATE, this.newMultiUriAwareCommandHandler({
+            execute: async uris => {
+                await Promise.all(uris.map(async uri => {
+                    const parent = await this.getParent(uri);
+                    if (parent) {
+                        const parentUri = new URI(parent.uri);
```

Comment:
How about getting `name` and `ext` from `Path`?
```ts
const parentUri = new URI(parent.uri);
const name = uri.path.name + '_copy';
const ext = uri.path.ext;
const target = this.findVacantChildUri(parentUri, parent, name, ext);
```

Reply:
Too easy. ;-)